src/Resources/contao/classes/Frontend.php

```diff
@@ -648,232 +648,4 @@ public static function indexPageIfApplicable(Response $objResponse)
 			}
 		}
 	}
-
-
-	/**
-	 * Check whether there is a cached version of the page and return a response object
-	 *
-	 * @return Response|null
-	 */
-	public static function getResponseFromCache()
```

Comment:
Well, from this point of view, 90% of the code base might be for internal use only. In the past we rejected event the slightest change in the API which could have been overridden/used by others and this includes all `public` and even most of the `protected` methods.
Keeping it as No-op and maybe commenting with "do not use, use ESI" in the phpDoc does not harm IMO but will not crash anything.